### PR TITLE
Feature/22 add x and y labels 

### LINKED ISSUE #22: Add wavelength units to the object
Wouldn't it be reasonable to add `##XUNITS=` as the label of wavelengths?

```r
file <- system.file("extdata", "SBO.jdx", package = "readJDX")
list_jdx <- readJDX(file)
list_jdx
```
image

e.g.:
```r
meta <- list_jdx$metadata
sub("^.*=", "", meta[grepl("XUNITS", meta)])
## [1] "1/CM"
```

What do you think, @bryanhanson?

### PR DESCRIPTION
**Pull request by GegznaV**
Closes #22 

I think, some unit tests should be added for this functionality, but I do not have time for this now. @bryanhanson, please review and give feedback.

### CODE REVIEW

**Review by bryanhanson — CHANGES_REQUESTED** [review]
Looks good overall. It looks like `jdx_extract_value` can be used not only as you have it, but also for CB's desire to extract arbitrary keys (in the long run).  However, I think the following line:
`key_pattern <- paste0("##\\$?", key, "=")`
should be changed to:
`key_pattern <- paste0("##\\$?", key, "=\\s*")`
to account for the fact that some vendors put variable amounts of space after the =.

In general this grep pattern will likely catch any of the standard fields along with some of the vendor fields, but I took a quick look at some of the 100+ JCAMP files I have in my test suite and there are other patterns in use -- many -- I would have to set aside a good block of time to figure out a more general approach, and this can wait.

**Comment by GegznaV** [comment]
> to account for the fact that some vendors put variable amounts of space after the =.

For that, I used `trimws()`

**Comment by GegznaV** [comment]
>  but I took a quick look at some of the 100+ JCAMP files I have in my test suite and there are other patterns in use 

Yes, I saw only 2-3 patterns. And this function is not general enough to account for all possible cases. I wrote the function by keeping `XUNITS` and `YUNITS` in mind only without looking to other possible usecases.

**Comment by GegznaV** [comment]
> should be changed to:
> `key_pattern <- paste0("##\\$?", key, "=\\s*")`

Done.

**Comment by GegznaV** [comment]
@bryanhanson, ready for review again.

**Comment by GegznaV** [comment]
The message of 6ef0ed2 fad to be `Update from devel`

File: R/read_jdx.R [hunk 1]
```diff
@@ -79,6 +74,10 @@ read_jdx <- function(file = stop("filename is needed"), SOFC = TRUE, debug = 0)
       packageDescription("hySpc.read.jdx")$BugReports
     )
   }
+
+  # Output
+  .spc_io_postprocess_optional(spc, filename = file)
+  # return(list(metadata = list_jdx[[2]], hyperSpec = spc))
```
Comment (cbeleites) [inline R/read_jdx.R]: please remove this commented-out line of code

**Review by cbeleites — CHANGES_REQUESTED** [review]

**Review by bryanhanson — APPROVED** [review]
With CB's note to remove the commented line, this looks good to me.

File: R/read_jdx.R (hunk 1 above)
Comment (GegznaV) [reply R/read_jdx.R]: I just moved the two rows outside the if-else statement. I'm not sure why the commented code was left. But, in general, I agree that there should not be any commented code left in the packages.

Comment (GegznaV) [reply R/read_jdx.R]: Done.

**Review by cbeleites — APPROVED** [review]

**Pull request merged**